// od-internal-header/esm2015/src/app/nav/topbar/topbar.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Component, Output, EventEmitter, Input } from '@angular/core';
import { HeaderService } from '../../services/header.service';
export class TopbarComponent {
    /**
     * @param {?} headerService
     */
    constructor(headerService) {
        this.headerService = headerService;
        this.sidenavToggle = new EventEmitter();
        this.environmentTitle = "";
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        this.getEnvironmentUrl(window.location.hostname);
    }
    /**
     * @return {?}
     */
    onToggleSidenav() {
        this.sidenavToggle.emit();
    }
    /**
     * @param {?} url
     * @return {?}
     */
    getEnvironmentUrl(url) {
        if (url.match('localhost')) {
            this.environmentTitle = 'LOCAL';
        }
        else if (url.endsWith('d.corp.odfl.com')) {
            this.environmentTitle = 'DEV';
        }
        else if (url.endsWith('l.corp.odfl.com')) {
            this.environmentTitle = 'LOAD';
        }
        else if (url.endsWith('q.corp.odfl.com')) {
            this.environmentTitle = 'QA';
        }
        else {
            this.environmentTitle = 'PROD';
        }
    }
}
TopbarComponent.decorators = [
    { type: Component, args: [{
                selector: 'od-internal-header-topbar',
                template: `<mat-toolbar color="primary">
    <div class="header">
      <img src="./assets/images/logos/OD_Logo.png">
    </div>

    <div fxFlex fxLayout fxFlexOffset="1%" fxLayoutGap="1%" fxLayoutAlign="flex-start center center">
        <div *ngFor="let page of headerService.pageList" fxHide.lt-md>
            <a routerLink={{page.routerLink}}>
                {{page.title}}
            </a>
        </div>
    </div>

    <div fxLayoutAlign="flex-end" fxLayoutGap="20px">
      <button mat-icon-button matTooltip="Version: 1.0.0">
       {{environmentTitle | uppercase}}
      </button>
  
        <button mat-icon-button (click)="onToggleSidenav()">
          <i class="fa fa-bars fa-2x"></i>
        </button>
    </div>

  </mat-toolbar>`,
                styles: [`mat-toolbar{background-color:#000}a{text-decoration:none;color:#fff;cursor:pointer}a:active,a:hover{color:#d3d3d3}.navigation-items{list-style:none;padding:0;margin:0}button,mat-icon{color:#fff}`]
            },] },
];
/** @nocollapse */
TopbarComponent.ctorParameters = () => [
    { type: HeaderService }
];
TopbarComponent.propDecorators = {
    sidenavToggle: [{ type: Output, args: ["sidenavToggle",] }],
    environmentTitle: [{ type: Input, args: ['environmentTitle',] }]
};
function TopbarComponent_tsickle_Closure_declarations() {
    /** @type {?} */
    TopbarComponent.prototype.sidenavToggle;
    /** @type {?} */
    TopbarComponent.prototype.environmentTitle;
    /** @type {?} */
    TopbarComponent.prototype.headerService;
}
