// od-internal-header/fesm5/od-internal-header.js
import { NgModule, Injectable, Component, EventEmitter, Output, Inject, Input } from '@angular/core';
import { CommonModule } from '@angular/common';
import { MatToolbarModule, MatSidenavModule, MatButtonModule, MatIconModule, MatListModule, MatMenuModule, MatTooltipModule, MatExpansionModule, MatDialogModule, MatFormFieldModule, MatInputModule, MatDialog, MatDialogRef, MAT_DIALOG_DATA } from '@angular/material';
import { RouterModule } from '@angular/router';
import { JoyrideService, JoyrideModule } from 'ngx-joyride';
import { FlexLayoutModule } from '@angular/flex-layout';
import { FormsModule } from '@angular/forms';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
var MaterialModule = /** @class */ (function () {
    function MaterialModule() {
    }
    MaterialModule.decorators = [
        { type: NgModule, args: [{
                    imports: [
                        CommonModule,
                        MatToolbarModule,
                        MatButtonModule,
                        MatSidenavModule,
                        MatIconModule,
                        MatListModule,
                        MatMenuModule,
                        MatTooltipModule,
                        MatExpansionModule,
                        MatDialogModule,
                        MatFormFieldModule,
                        MatInputModule
                    ],
                    exports: [
                        MatToolbarModule,
                        MatSidenavModule,
                        MatButtonModule,
                        MatIconModule,
                        MatListModule,
                        MatMenuModule,
                        MatTooltipModule,
                        MatExpansionModule,
                        MatDialogModule,
                        MatFormFieldModule,
                        MatInputModule
                    ],
                    declarations: []
                },] },
    ];
    return MaterialModule;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
var /** @type {?} */ routes = [];
var AppRoutingModule = /** @class */ (function () {
    function AppRoutingModule() {
    }
    AppRoutingModule.decorators = [
        { type: NgModule, args: [{
                    imports: [
                        RouterModule.forRoot(routes)
                    ],
                    exports: [RouterModule],
                    declarations: []
                },] },
    ];
    return AppRoutingModule;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
var ODJoyrideService = /** @class */ (function () {
    function ODJoyrideService() {
        this.steps = [];
    }
    ODJoyrideService.decorators = [
        { type: Injectable },
    ];
    /** @nocollapse */
    ODJoyrideService.ctorParameters = function () { return []; };
    return ODJoyrideService;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
var HeaderService = /** @class */ (function () {
    function HeaderService() {
        this.applicationNavList = [];
        this.pageList = [];
    }
    HeaderService.decorators = [
        { type: Injectable },
    ];
    /** @nocollapse */
    HeaderService.ctorParameters = function () { return []; };
    return HeaderService;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
var SidenavListComponent = /** @class */ (function () {
    function SidenavListComponent(joyrideService, dialog, oDJoyrideService, headerService) {
        this.joyrideService = joyrideService;
        this.dialog = dialog;
        this.oDJoyrideService = oDJoyrideService;
        this.headerService = headerService;
        this.closeSidenav = new EventEmitter();
    }
    /**
     * @return {?}
     */
    SidenavListComponent.prototype.onClose = /**
     * @return {?}
     */
    function () {
        this.closeSidenav.emit();
    };
    /**
     * @return {?}
     */
    SidenavListComponent.prototype.startJoyride = /**
     * @return {?}
     */
    function () {
        this.onClose();
        this.joyrideService.startTour({
            steps: this.oDJoyrideService.steps,
            stepDefaultPosition: 'bottom'
        } // Your steps order
        );
    };
    /**
     * @return {?}
     */
    SidenavListComponent.prototype.openServiceTicketDialog = /**
     * @return {?}
     */
    function () {
        this.dialog.open(ServiceTicketDialog, {
            width: '500px'
        });
    };
    SidenavListComponent.decorators = [
        { type: Component, args: [{
                    selector: 'od-internal-header-sidenav-list',
                    template: "<mat-nav-list>\n  <a mat-list-item (click)=\"onClose()\">\n    <mat-icon>\n      arrow_forward_ios\n    </mat-icon>\n    <span class=\"nav-caption\">Close</span>\n  </a>\n\n  <mat-accordion>\n    <mat-expansion-panel fxHide.gt-sm *ngIf=\"headerService.pageList.length > 0\">\n      <mat-expansion-panel-header>\n        Links\n      </mat-expansion-panel-header>\n      <div *ngFor=\"let page of headerService.pageList\">\n          <a routerLink={{page.routerLink}} mat-list-item (click)=\"onClose()\">\n              <span class=\"nav-caption \">\n              {{page.title}}\n              </span>\n          </a>\n      </div>\n    </mat-expansion-panel>\n\n    <mat-expansion-panel *ngIf=\"headerService.applicationNavList.length > 0\">\n      <mat-expansion-panel-header>\n        <mat-panel-title>\n          {{headerService.applicationGroupTitle}}\n        </mat-panel-title>\n      </mat-expansion-panel-header>\n      <div *ngFor=\"let app of headerService.applicationNavList\">\n\n        <a href=\"{{app.url}}\" mat-list-item (click)=\"onClose()\">\n\n          <span class=\"nav-caption \">\n            <span class=\"fa-stack\">\n              <div *ngFor=\"let icon of app.icons\">\n                <i class=\"fa {{icon}}\"></i>\n              </div>\n            </span>\n            {{app.title}}</span>\n        </a>\n      </div>\n    </mat-expansion-panel>\n\n    <mat-expansion-panel>\n      <mat-expansion-panel-header>\n        <mat-panel-title>\n          Help\n        </mat-panel-title>\n      </mat-expansion-panel-header>\n      <a mat-list-item (click)=\"openServiceTicketDialog()\">\n        <mat-icon>local_hospital</mat-icon>\n        Service Desk Ticket\n      </a>\n\n      <a mat-list-item (click)=\"startJoyride()\">\n        <mat-icon>child_care</mat-icon>\n        Tutorial\n      </a>\n    </mat-expansion-panel>\n  </mat-accordion>\n</mat-nav-list>",
                    styles: ["a{text-decoration:none;color:#fff}a:active,a:hover{color:#d3d3d3}.nav-caption{display:inline-block;padding-left:6px}.mat-nav-list .mat-list-item:hover,mat-expansion-panel:hover{color:#fff}"]
                },] },
    ];
    /** @nocollapse */
    SidenavListComponent.ctorParameters = function () { return [
        { type: JoyrideService },
        { type: MatDialog },
        { type: ODJoyrideService },
        { type: HeaderService }
    ]; };
    SidenavListComponent.propDecorators = {
        closeSidenav: [{ type: Output, args: ["closeSidenav",] }]
    };
    return SidenavListComponent;
}());
var ServiceTicketDialog = /** @class */ (function () {
    function ServiceTicketDialog(dialogRef, data) {
        this.dialogRef = dialogRef;
        this.data = data;
    }
    /**
     * @return {?}
     */
    ServiceTicketDialog.prototype.onNoClick = /**
     * @return {?}
     */
    function () {
        this.dialogRef.close();
    };
    /**
     * @param {?} form
     * @return {?}
     */
    ServiceTicketDialog.prototype.onSubmit = /**
     * @param {?} form
     * @return {?}
     */
    function (form) {
        console.log(form);
    };
    ServiceTicketDialog.decorators = [
        { type: Component, args: [{
                    selector: 'service-ticket-dialog',
                    template: "<div class=\"container\" fxLayout=\"column\"  fxLayoutWrap fxLayoutGap=\"10px\" fxLayoutAlign=\"center stretch\">\n    <div fxLayoutAlign=\"center center\">\n        <h2>Service Desk Ticket</h2>\n    </div>\n    <div>\n        <form #f=\"ngForm\" (ngSubmit)=\"onSubmit(f)\">\n            <div fxLayoutAlign=\"center stretch\">\n            <mat-form-field fxFlex=\"100%\">\n                <textarea name=\"serviceArea\" ngModel matInput placeholder=\"Please enter your questions or comments\" required></textarea>\n            </mat-form-field>\n            </div>\n            <div fxLayoutAlign=\"center center\">\n                <button type=\"submit\" mat-raised-button color=\"primary\" [disabled]=\"f.invalid\">Submit</button>\n            </div>\n        </form>\n    </div>\n</div>",
                },] },
    ];
    /** @nocollapse */
    ServiceTicketDialog.ctorParameters = function () { return [
        { type: MatDialogRef },
        { type: undefined, decorators: [{ type: Inject, args: [MAT_DIALOG_DATA,] }] }
    ]; };
    return ServiceTicketDialog;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
var HeaderComponent = /** @class */ (function () {
    function HeaderComponent() {
    }
    HeaderComponent.decorators = [
        { type: Component, args: [{
                    selector: 'header',
                    template: "\n    <mat-sidenav #sidenav role=\"navigation\" position=\"end\">\n        <od-internal-header-sidenav-list (closeSidenav)=\"sidenav.close()\"></od-internal-header-sidenav-list>\n    </mat-sidenav>\n    <mat-sidenav-content role=\"main\">\n       <od-internal-header-topbar (sidenavToggle)=\"sidenav.toggle()\"></od-internal-header-topbar>\n        <router-outlet></router-outlet>\n    </mat-sidenav-content>\n\n",
                    styles: [""]
                },] },
    ];
    /** @nocollapse */
    HeaderComponent.ctorParameters = function () { return []; };
    return HeaderComponent;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
var TopbarComponent = /** @class */ (function () {
    function TopbarComponent(headerService) {
        this.headerService = headerService;
        this.sidenavToggle = new EventEmitter();
        this.environmentTitle = "";
    }
    /**
     * @return {?}
     */
    TopbarComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        this.getEnvironmentUrl(window.location.hostname);
    };
    /**
     * @return {?}
     */
    TopbarComponent.prototype.onToggleSidenav = /**
     * @return {?}
     */
    function () {
        this.sidenavToggle.emit();
    };
    /**
     * @param {?} url
     * @return {?}
     */
    TopbarComponent.prototype.getEnvironmentUrl = /**
     * @param {?} url
     * @return {?}
     */
    function (url) {
        if (url.match('localhost')) {
            this.environmentTitle = 'LOCAL';
        }
        else if (url.endsWith('d.corp.odfl.com')) {
            this.environmentTitle = 'DEV';
        }
        else if (url.endsWith('l.corp.odfl.com')) {
            this.environmentTitle = 'LOAD';
        }
        else if (url.endsWith('q.corp.odfl.com')) {
            this.environmentTitle = 'QA';
        }
        else {
            this.environmentTitle = 'PROD';
        }
    };
    TopbarComponent.decorators = [
        { type: Component, args: [{
                    selector: 'od-internal-header-topbar',
                    template: "<mat-toolbar color=\"primary\">\n    <div class=\"header\">\n      <img src=\"./assets/images/logos/OD_Logo.png\">\n    </div>\n\n    <div fxFlex fxLayout fxFlexOffset=\"1%\" fxLayoutGap=\"1%\" fxLayoutAlign=\"flex-start center center\">\n        <div *ngFor=\"let page of headerService.pageList\" fxHide.lt-md>\n            <a routerLink={{page.routerLink}}>\n                {{page.title}}\n            </a>\n        </div>\n    </div>\n\n    <div fxLayoutAlign=\"flex-end\" fxLayoutGap=\"20px\">\n      <button mat-icon-button matTooltip=\"Version: 1.0.0\">\n       {{environmentTitle | uppercase}}\n      </button>\n  \n        <button mat-icon-button (click)=\"onToggleSidenav()\">\n          <i class=\"fa fa-bars fa-2x\"></i>\n        </button>\n    </div>\n\n  </mat-toolbar>",
                    styles: ["mat-toolbar{background-color:#000}a{text-decoration:none;color:#fff;cursor:pointer}a:active,a:hover{color:#d3d3d3}.navigation-items{list-style:none;padding:0;margin:0}button,mat-icon{color:#fff}"]
                },] },
    ];
    /** @nocollapse */
    TopbarComponent.ctorParameters = function () { return [
        { type: HeaderService }
    ]; };
    TopbarComponent.propDecorators = {
        sidenavToggle: [{ type: Output, args: ["sidenavToggle",] }],
        environmentTitle: [{ type: Input, args: ['environmentTitle',] }]
    };
    return TopbarComponent;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
var HeaderModule = /** @class */ (function () {
    function HeaderModule() {
    }
    HeaderModule.decorators = [
        { type: NgModule, args: [{
                    imports: [
                        CommonModule,
                        MaterialModule,
                        FlexLayoutModule,
                        AppRoutingModule,
                        JoyrideModule.forRoot(),
                        FormsModule
                    ],
                    exports: [TopbarComponent, SidenavListComponent, HeaderComponent, ServiceTicketDialog],
                    entryComponents: [
                        ServiceTicketDialog
                    ],
                    providers: [ODJoyrideService, HeaderService],
                    declarations: [TopbarComponent, SidenavListComponent, HeaderComponent, ServiceTicketDialog]
                },] },
    ];
    return HeaderModule;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
var Application = /** @class */ (function () {
    function Application() {
    }
    return Application;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
var Page = /** @class */ (function () {
    function Page() {
    }
    return Page;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */

export { HeaderModule, ODJoyrideService, HeaderService, Application, Page, AppRoutingModule as ɵb, MaterialModule as ɵa, HeaderComponent as ɵf, ServiceTicketDialog as ɵe, SidenavListComponent as ɵd, TopbarComponent as ɵc };
