// od-internal-header/esm5/src/app/nav/header/header.module.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { MaterialModule } from '../../material/material.module';
import { FlexLayoutModule } from '@angular/flex-layout';
import { AppRoutingModule } from '../../app-routing/app-routing.module';
import { SidenavListComponent, ServiceTicketDialog } from '../sidenav-list/sidenav-list.component';
import { HeaderComponent } from './header.component';
import { JoyrideModule } from 'ngx-joyride';
import { TopbarComponent } from '../topbar/topbar.component';
import { FormsModule } from '@angular/forms';
import { ODJoyrideService } from '../../services/od-joyride.service';
import { HeaderService } from '../../services/header.service';
var HeaderModule = /** @class */ (function () {
    function HeaderModule() {
    }
    HeaderModule.decorators = [
        { type: NgModule, args: [{
                    imports: [
                        CommonModule,
                        MaterialModule,
                        FlexLayoutModule,
                        AppRoutingModule,
                        JoyrideModule.forRoot(),
                        FormsModule
                    ],
                    exports: [TopbarComponent, SidenavListComponent, HeaderComponent, ServiceTicketDialog],
                    entryComponents: [
                        ServiceTicketDialog
                    ],
                    providers: [ODJoyrideService, HeaderService],
                    declarations: [TopbarComponent, SidenavListComponent, HeaderComponent, ServiceTicketDialog]
                },] },
    ];
    return HeaderModule;
}());
export { HeaderModule };
