// od-internal-header/esm5/src/app/nav/topbar/topbar.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Component, Output, EventEmitter, Input } from '@angular/core';
import { HeaderService } from '../../services/header.service';
var TopbarComponent = /** @class */ (function () {
    function TopbarComponent(headerService) {
        this.headerService = headerService;
        this.sidenavToggle = new EventEmitter();
        this.environmentTitle = "";
    }
    /**
     * @return {?}
     */
    TopbarComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        this.getEnvironmentUrl(window.location.hostname);
    };
    /**
     * @return {?}
     */
    TopbarComponent.prototype.onToggleSidenav = /**
     * @return {?}
     */
    function () {
        this.sidenavToggle.emit();
    };
    /**
     * @param {?} url
     * @return {?}
     */
    TopbarComponent.prototype.getEnvironmentUrl = /**
     * @param {?} url
     * @return {?}
     */
    function (url) {
        if (url.match('localhost')) {
            this.environmentTitle = 'LOCAL';
        }
        else if (url.endsWith('d.corp.odfl.com')) {
            this.environmentTitle = 'DEV';
        }
        else if (url.endsWith('l.corp.odfl.com')) {
            this.environmentTitle = 'LOAD';
        }
        else if (url.endsWith('q.corp.odfl.com')) {
            this.environmentTitle = 'QA';
        }
        else {
            this.environmentTitle = 'PROD';
        }
    };
    TopbarComponent.decorators = [
        { type: Component, args: [{
                    selector: 'od-internal-header-topbar',
                    template: "<mat-toolbar color=\"primary\">\n    <div class=\"header\">\n      <img src=\"./assets/images/logos/OD_Logo.png\">\n    </div>\n\n    <div fxFlex fxLayout fxFlexOffset=\"1%\" fxLayoutGap=\"1%\" fxLayoutAlign=\"flex-start center center\">\n        <div *ngFor=\"let page of headerService.pageList\" fxHide.lt-md>\n            <a routerLink={{page.routerLink}}>\n                {{page.title}}\n            </a>\n        </div>\n    </div>\n\n    <div fxLayoutAlign=\"flex-end\" fxLayoutGap=\"20px\">\n      <button mat-icon-button matTooltip=\"Version: 1.0.0\">\n       {{environmentTitle | uppercase}}\n      </button>\n  \n        <button mat-icon-button (click)=\"onToggleSidenav()\">\n          <i class=\"fa fa-bars fa-2x\"></i>\n        </button>\n    </div>\n\n  </mat-toolbar>",
                    styles: ["mat-toolbar{background-color:#000}a{text-decoration:none;color:#fff;cursor:pointer}a:active,a:hover{color:#d3d3d3}.navigation-items{list-style:none;padding:0;margin:0}button,mat-icon{color:#fff}"]
                },] },
    ];
    /** @nocollapse */
    TopbarComponent.ctorParameters = function () { return [
        { type: HeaderService }
    ]; };
    TopbarComponent.propDecorators = {
        sidenavToggle: [{ type: Output, args: ["sidenavToggle",] }],
        environmentTitle: [{ type: Input, args: ['environmentTitle',] }]
    };
    return TopbarComponent;
}());
export { TopbarComponent };
function TopbarComponent_tsickle_Closure_declarations() {
    /** @type {?} */
    TopbarComponent.prototype.sidenavToggle;
    /** @type {?} */
    TopbarComponent.prototype.environmentTitle;
    /** @type {?} */
    TopbarComponent.prototype.headerService;
}
